extension-style/src/main/java/com/mapbox/maps/extension/style/types/PromoteId.kt

```diff
@@ -0,0 +1,47 @@
+package com.mapbox.maps.extension.style.types
+
+import androidx.annotation.Keep
+import com.mapbox.bindgen.Value
+import com.mapbox.maps.extension.style.sources.generated.GeoJsonSource
+import com.mapbox.maps.extension.style.sources.generated.VectorSource
+
+/**
+ * Holds a property type to promote a specific feature for feature state API.
+ *
+ * @param sourceId source layer id of the feature, either source [GeoJsonSource] or [VectorSource].
+ * @param propertyName feature property name.
+ *
+ * For more information see [The online documentation]https://docs.mapbox.com/mapbox-gl-js/style-spec/types/#promoteId).
+ */
+@Keep
+data class PromoteId @JvmOverloads constructor(
+  val sourceId: String? = null,
+  val propertyName: String
+) {
+  internal fun toValue(): Value {
+    sourceId?.let {
+      return Value(hashMapOf(sourceId to Value(propertyName)))
+    }
+    return Value(propertyName)
+  }
+
+  companion object {
+    /**
+     * Construct a [PromoteId] object from a Property returned from the core.
+     * Can be either of type [String] or [HashMap] of <String,String>
+     */
+    fun fromProperty(propertyName: Any) = when (propertyName) {
```

Comment:
Could you add tests for this method and cover all scenarios?

Reply:
I think we could split this function to two functions to make the API signature strongly typed, since we only support two types of data.  i.e. `fromProperty(propertyName: String)` and `fromProperty(propertyMap: Map<String, String>)`. 

Thinking again, why would we need to have this method? The promoteId could either a property name, or an object of the form {<sourceLayer>: <propertyName>}, so the constructor should be well enough for this purpose.

Reply:
> I think we could split this function to two functions to make the API signature strongly typed, since we only support two types of data. i.e. `fromProperty(propertyName: String)` and `fromProperty(propertyMap: Map<String, String>)`.
> 
> Thinking again, why would we need to have this method? The promoteId could either a property name, or an object of the form {: }, so the constructor should be well enough for this purpose. cc @ank27

It was made this way to handle the errors in case core provides other type formats. 
hmm, I see that we can use `Value` as it is already wrapper for type.